x-pack/dockerlogbeat/pipelinemock/reader.go

```diff
@@ -0,0 +1,50 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package pipelinemock
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io"
+	"io/ioutil"
+	"testing"
+
+	"github.com/docker/docker/api/types/plugins/logdriver"
+	"github.com/gogo/protobuf/proto"
+	"github.com/stretchr/testify/assert"
+)
+
+// CreateTestInputFromLine returns a ReadCloser based on an input string
+func CreateTestInputFromLine(t *testing.T, line string) io.ReadCloser {
+	exampleStruct := &logdriver.LogEntry{
+		Source:   "Test",
+		TimeNano: 0,
+		Line:     []byte(line),
+		Partial:  false,
+		PartialLogMetadata: &logdriver.PartialLogEntryMetadata{
+			Last:    false,
+			Id:      "",
+			Ordinal: 0,
+		},
+	}
+
+	writer := new(bytes.Buffer)
+
+	encodeLog(t, writer, exampleStruct)
+	return ioutil.NopCloser(writer)
+}
+
+func encodeLog(t *testing.T, out io.Writer, entry *logdriver.LogEntry) {
+	rawBytes, err := proto.Marshal(entry)
+	assert.NoError(t, err)
+
+	sizeBytes := make([]byte, 4)
+	binary.BigEndian.PutUint32(sizeBytes, uint32(len(rawBytes)))
+
+	_, err = out.Write(sizeBytes)
+	assert.NoError(t, err)
+	_, err = out.Write(rawBytes)
+	assert.NoError(t, err)
```

Comment:
better use `require.NoError`. The `assert` package allows the test to continue with potentially faulty data, adding loads of noise to the test output. Calling `require.NoError` triggers `t.Fatal`, that will stop the test immediately.